services/wireguard/endpoint/netstack-provider/shaper_service.go

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2022 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package netstack_provider
+
+import (
+	"github.com/mysteriumnetwork/node/config"
+	"github.com/mysteriumnetwork/node/eventbus"
+	"github.com/rs/zerolog/log"
+	"golang.org/x/time/rate"
+)
+
+var rateLimiter *rate.Limiter
+
+func getRateLimitter() *rate.Limiter {
+	return rateLimiter
+}
+
+func InitUserspaceShaper(eventBus eventbus.EventBus) {
+
+	applyLimits := func(e interface{}) {
+		bandwidthBytes := config.GetUInt64(config.FlagShaperBandwidth) * 1024
+		bandwidth := rate.Limit(bandwidthBytes)
+		if !config.GetBool(config.FlagShaperEnabled) {
+			bandwidth = rate.Inf
+		}
+		log.Warn().Msgf("Shaper bandwidth: %v", config.GetUInt64(config.FlagShaperBandwidth))
+		rateLimiter.SetLimit(bandwidth)
+	}
+
+	bandwidthBytes := config.GetUInt64(config.FlagShaperBandwidth) * 1024
+	bandwidth := rate.Limit(bandwidthBytes)
+	log.Warn().Msgf("Shaper bandwidth: %v", config.GetUInt64(config.FlagShaperBandwidth))
```

Comment:
All of this looks very similar to `applyLimits` function, maybe we can change it a bit and avoid code duplication?

Reply:
ok